modules/renter/persist.go

```diff
@@ -270,13 +272,17 @@ func (r *Renter) shareFiles(nicknames []string, w io.Writer) error {
 		return err
 	}
 
+	// Create compressor.
+	zip, _ := gzip.NewWriterLevel(w, gzip.BestCompression)
+	defer zip.Close()
+
 	// Write each file.
 	for _, name := range nicknames {
 		file, exists := r.files[name]
 		if !exists {
 			return ErrUnknownPath
 		}
-		err := file.save(w)
+		err := file.save(zip)
```

Comment:
it's not clear from the comments and documentation that 'save' is meant to be chained into the same file or compression scheme. I think that it makes sense to change the name, perhaps to `pack` or `write` or something, and document that it's meant to be chained.

Logic looks good, just not well explained enough.